client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -250,6 +256,11 @@ class SimplePaymentsDialog extends Component {
 		} );
 	};
 
+	addNewPaymentButton = () => {
+		this.setState( { activeTab: 'form', editedPaymentId: null } );
+		this.formStateController.resetFields( this.getInitialFormFields( null ) );
```

Comment:
New method was added to distinguish edit operation from addNew. Again, explicit over implicit.